File: Common/TTS Test/tts_client.py
import asyncio
import base64
import json
import subprocess
import sys
from pathlib import Path

import websockets


URI = "ws://96.32.24.78:8080"

SRS_EXTERNAL_AUDIO_EXE = Path(
    r"E:\DCS-SimpleRadioStandalone-2.0.8.5\ExternalAudio\DCS-SR-ExternalAudio.exe"
)


async def main():
    text = sys.argv[1] if len(sys.argv) > 1 else "hello from python"
    output_dir = Path(sys.argv[2]) if len(sys.argv) > 2 else Path(".")

    output_dir.mkdir(parents=True, exist_ok=True)

    async with websockets.connect(URI) as websocket:
        await websocket.send(text)
        message = await websocket.recv()

    data = json.loads(message)

    if not data.get("success"):
        raise RuntimeError(data.get("error", "unknown server error"))

    audio = base64.b64decode(data["audio"])
    ext = data.get("format", "ogg")
    output_file = output_dir / f"output.{ext}"

    output_file.write_bytes(audio)

    print(output_file)

    subprocess.run(
        [
            str(SRS_EXTERNAL_AUDIO_EXE),
            f"-i={str(output_file)}",
            "--freqs=250.0",
            "--modulations=AM",
            "-g=male",
            "--coalition=2",
        ],
        check=True,
    )


if __name__ == "__main__":
    asyncio.run(main())

File: Common/TTS Test/tts_client.lua
local lua_modules = [[C:\Users\fpere\IdeaProjects\NASGroupMissionScripts\Common\TTS Test\lua_modules]]

package.path =
lua_modules .. [[\?.lua;]] ..
        lua_modules .. [[\?\init.lua;]] ..
        lua_modules .. [[\share\lua\5.1\?.lua;]] ..
        lua_modules .. [[\share\lua\5.1\?\init.lua;]] ..
        package.path

package.cpath =
lua_modules .. [[\?.dll;]] ..
        lua_modules .. [[\lib\lua\5.1\?.dll;]] ..
        package.cpath

local websocket = require("websocket.client")
local mime = require("mime")
local json = require("dkjson")

local url = "ws://96.32.24.78:8080"

local ws = websocket()

local ok, err = ws:connect(url)
if not ok then
  error("WebSocket connect failed: " .. tostring(err))
end

ws:send("hello from lua")

local response_body = ws:receive()
ws:close()

if not response_body then
  error("No response from server")
end

local data = json.decode(response_body)

if not data then
  error("Failed to parse JSON response: " .. tostring(response_body))
end

if data.success and data.audio then
  local ext = data.format or "ogg"
  local audio_data = mime.unb64(data.audio)
  local filename = "output." .. ext

  local f = assert(io.open(filename, "wb"))
  f:write(audio_data)
  f:close()

  print("Saved audio to", filename)
else
  print("Server error:", data.error or "unknown")
end